src/classes/THAN_Filter_TDTM.cls

```diff
@@ -61,12 +61,16 @@ public class THAN_Filter_TDTM extends TDTM_Runnable {
 	                    String filterField = String.escapeSingleQuotes(handler.Filter_Field__c);
 	                    String obj = String.escapeSingleQuotes(handler.Object__c);
 	                    if(!String.isBlank(filterField) && !String.isBlank(obj)) {
-		                    try {
-		                        String dynamicQuery = 'select ' + filterField + ' from ' + obj + ' limit 1';
-		                        List<SObject> records =  Database.query(dynamicQuery);
-		                    } catch(QueryException e) {
-		                        handler.addError(Label.InvalidFilter + handler.Class__c);
-		                    }
+                            try {
+                                //Note: We should use schema class to check if the field is valid for an object since dynamic query will ignore namespace automatically.
+                                SObjectField field = Schema.getGlobalDescribe().get(obj).getDescribe().fields.getMap().get(filterField);
+                                //Note: System will automatically remove HEDA namespace of client's custom field. For example, client create a custom field ABC__c, and put hed__ABC__c into the setting. The field value will not be null but ABC__c. We do not want user to accidentally add namespace into their own field.
+                                if(field == null || String.valueOf(field) != filterField) {
```

Comment:
Thanks for the comment in line 67. I was going to ask you about the second condition of the if statement, but now I don't need to. :-)

Reply:
Now that there's no query, do we still need the try-catch? No harm in having it, but it's possible nothing has now the potential of throwing an exception in this section, and we would avoid having the handler.addError line twice. Just a thought.

Reply:
Yes, I think it is correct. I just did a little bit test, looks ok. So, I just removed it.